### PR TITLE
Expose Assembly.EntryPoint via public contract

### LINKED ISSUE #3336: Expose Assembly.EntryPoint
Today Assembly is missing EntryPoint, it would be great if this were exposed (as it in on .NET Framework) so that managed code can invoke entry points in other managed executables.

See https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Dnx.Runtime.Sources/Impl/EntryPointExecutor.cs#L77 as an example.

/cc @nguerrera

### PR DESCRIPTION
**Pull request by nguerrera**
Fix #3336

### CODE REVIEW

File: src/System.Reflection/tests/TestExe/Program.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    public class TestExeException : Exception
+    {
+    }
+
+    public static class TestExe
+    {
+        private static void Main()
+        {
+            throw new TestExeException();
+        }
```
Comment (stephentoub) [inline src/System.Reflection/tests/TestExe/Program.cs]: Should we also validate invoking entry points that return Int32 values?

**Comment by stephentoub** [comment]
One suggestion on tests, otherwise LGTM.

File: src/System.Reflection/tests/TestExe/Program.cs (hunk 1 above)
Comment (nguerrera) [inline src/System.Reflection/tests/TestExe/Program.cs]: I switched it to int32, not sure why I had void on the brain. I don't think it's too bad to only test one signature. I'd have to create 1 test exe per signature, which seems like overkill.

Comment (stephentoub) [inline src/System.Reflection/tests/TestExe/Program.cs]: Ok, sounds good.  If you wanted to test the exception behavior as well, you could change the signature to take a string[] args, conditionally throw or return based on some input, and then have multiple test cases with the same exe, passing in different inputs.  But it's fine as is.

**Comment by mellinoe** [comment]
LGTM

**Comment by davidfowl** [comment]
SHIP IT YESTERDAY!!! :ship: :it: :shipit: !!!!!!

**Comment by davidfowl** [comment]
😀

**Pull request merged**